### PR TITLE
If token based authentication is mandatory a login failed entry is wr…

### PR DESCRIPTION
**Pull request by DeepDiver1975**
…itten to the log so that fail2ban can handle this case.

## Description
A log message is written in case a user does not login using an app password if 'token_auth_enforced' is set to true

## Related Issue
https://github.com/owncloud/enterprise/issues/2473

## How Has This Been Tested?
- unit tests executed
- manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: lib/private/User/Session.php [hunk 1]
```diff
@@ -341,6 +342,7 @@ public function logClientIn($user, $password, IRequest $request) {
 			throw new \InvalidArgumentException('$user cannot be empty');
 		}
 		if (!$isTokenPassword && $this->isTokenAuthEnforced()) {
+			$this->logger->warning("Login failed: '$user' (Remote IP: '{$request->getRemoteAddress()}')", ['app' => 'core']);
```
[the remark this answers is not in the document]
Comment (sharidas) [reply lib/private/User/Session.php]: Yes the login failures are addressed https://github.com/owncloud/core/pull/31467. This method https://github.com/owncloud/core/blob/master/lib/private/User/Session.php#L1078 is called for the same.

[the remark this answers is not in the document]
Comment (DeepDiver1975) [reply lib/private/User/Session.php]: will emit the event ..... gimme a sec ...

**Comment by phil-davis** [comment]
Tested against this branch just now.

- with 10.0.9RC2 and log level 0 the ``PasswordLoginForbiddenException`` is logged, but no "Login failed" in the log. This is the existing mis-behavior.
- with this branch and log level 0 (debug) both "Login failed" and the ``PasswordLoginForbiddenException`` messages are logged.
- with this branch and log level 2 (warning) the "Login failed" message only is logged.
- with this branch and log level 3 (error) no message is logged (login fail is only warning level).

Works for me.

**Review by phil-davis — APPROVED** [review]
Tested as per my comment.
Others have reviewed the actual code.

**Pull request merged**